synfig-studio/src/gui/app.cpp

```diff
@@ -2312,7 +2319,8 @@ App::quit()
 			return;
 	process_all_events();
 
-	Gtk::Main::quit();
+	// Gtk::Main::quit();
+	app_reference->remove_window(*main_window);
```

Comment:
In case of `singleton` model you should use `App::instance()` here.

Reply:
![image](https://user-images.githubusercontent.com/71627983/127769782-0b9b1571-12d7-4ec5-bd00-15fc17cdc9b1.png)

I think instance is deprecated if I am not wrong . Is there any alternative to it ?

Reply:
He suggested you to rename the method `create()` you… created… to `instance()` to be consistent with the other cases Synfig code do it.

Reply:
Okie okie ! Sorry my bad ! Understood ! 🙂

Reply:
Or not. Maybe I misunderstood him!

Reply:
@ice0 a long time ago, someone made studio::App a 'static class' (every property and method is static). So I don't think we need to do the "instance()" method. We could change it later, but not on this GSoC, due time running.

Reply:
> a long time ago, someone made studio::App a 'static class' (every property and method is static). 

I missed that moment. Anyway, I suggest not refactoring the `App` class, but just rename the `create()` method to make it clearer. If I know that class is a Singleton, I will always look for `instance()` or `get_instance()` methods, so looking for public members like `app_reference` or something is not clear to me.

Reply:
> He suggested you to rename the method create() you… created… to instance() to be consistent with the other cases Synfig code do it.

Yes. So in that case you need to use `App::instance()->remove_window(*main_window)` instead of `app_reference->remove_window(*main_window)`